File: flutter_app/lib/features/orders/domain/order.dart
class OrderItem {
  const OrderItem({
    required this.menuItemId,
    required this.name,
    required this.quantity,
    required this.unitPrice,
  });

  final int menuItemId;
  final String name;
  final int quantity;
  final double unitPrice;

  factory OrderItem.fromJson(Map<String, dynamic> j) => OrderItem(
        menuItemId: j['menu_item_id'] as int,
        name: j['name'] as String,
        quantity: j['quantity'] as int,
        unitPrice: (j['unit_price'] as num).toDouble(),
      );
}

class Order {
  const Order({
    required this.id,
    required this.shopId,
    required this.items,
    required this.totalPrice,
    required this.status,
    required this.address,
    required this.createdAt,
  });

  final String id;
  final int shopId;
  final List<OrderItem> items;
  final double totalPrice;
  final String status;
  final String address;
  final DateTime createdAt;

  String get statusLabel => switch (status) {
        'pending' => 'Ожидает',
        'accepted' => 'Принят',
        'picked_up' => 'В пути',
        'delivered' => 'Доставлен',
        'cancelled' => 'Отменён',
        _ => status,
      };

  factory Order.fromJson(Map<String, dynamic> j) => Order(
        id: j['id'].toString(),
        shopId: j['shop_id'] as int,
        items: (j['items'] as List<dynamic>? ?? [])
            .map((e) => OrderItem.fromJson(e as Map<String, dynamic>))
            .toList(),
        totalPrice: (j['total_price'] as num).toDouble(),
        status: j['status'] as String,
        address: (j['address'] as String?) ?? '',
        createdAt: DateTime.parse(j['created_at'] as String),
      );
}

File: order-service/internal/models/order.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type OrderStatus string

const (
	StatusPending   OrderStatus = "pending"
	StatusAccepted  OrderStatus = "accepted"
	StatusPickedUp  OrderStatus = "picked_up"
	StatusDelivered OrderStatus = "delivered"
	StatusCancelled OrderStatus = "cancelled"
)

type Order struct {
	ID         uint           `gorm:"primarykey" json:"id"`
	CustomerID uint           `gorm:"not null;index" json:"customer_id"`
	ShopID     uint           `gorm:"not null;index" json:"shop_id"`
	CourierID  *uint          `json:"courier_id"`
	Status     OrderStatus    `gorm:"type:varchar(20);not null;default:'pending'" json:"status"`
	TotalPrice float64        `gorm:"not null" json:"total_price"`
	Address    string         `gorm:"not null" json:"address"`
	Items      []OrderItem    `gorm:"foreignKey:OrderID" json:"items"`
	CreatedAt  time.Time      `json:"created_at"`
	UpdatedAt  time.Time      `json:"updated_at"`
	DeletedAt  gorm.DeletedAt `gorm:"index" json:"-"`
}

type OrderItem struct {
	ID         uint    `gorm:"primarykey" json:"id"`
	OrderID    uint    `gorm:"not null;index" json:"order_id"`
	MenuItemID uint    `gorm:"not null" json:"menu_item_id"`
	Name       string  `gorm:"not null" json:"name"`
	UnitPrice  float64 `gorm:"not null" json:"unit_price"`
	Quantity   int     `gorm:"not null" json:"quantity"`
}

type OrderDecline struct {
	OrderID   uint `gorm:"not null;index;uniqueIndex:idx_order_courier_decline" json:"order_id"`
	CourierID uint `gorm:"not null;index;uniqueIndex:idx_order_courier_decline" json:"courier_id"`
}
